io-engine/src/grpc/v1/pool.rs

```diff
@@ -57,6 +67,48 @@ impl From<GrowPoolRequest> for FindPoolArgs {
     }
 }
 
+/// Helper routine to extract Encryption params from the Create or Import pool request.
+async fn util_fetch_secret_params(
+    params: &PoolEncryptionParams,
+) -> Result<Option<PoolEncKey>, Status> {
+    let enc_key = match params {
+        PoolEncryptionParams::Create(enc_arg) => {
+            match enc_arg.clone() {
+                // Would have been nice to deduplicate the code if we could call secret_data using a
+                // trait object returned by library. But it doesn't seem to work here because the
+                // library call having generic type in definition isn't allowing that.
+                create_pool_request::Encryption::Data(ckd) => {
+                    Some(ckd.try_into().map_err(|_| {
+                        Status::invalid_argument("Couldn't parse raw secret params".to_string())
```

Comment:
Is the returned error from try_into useful, if so we could either use it fully or make it mart of the message

```suggestion
                    Some(ckd.try_into().map_err(|error| {
                        Status::invalid_argument(error.to_string())
```